### ISSUE TITLE
Tons of memory used when refining VRMarkers

### ISSUE THREAD

**Issue by thetheodor** [DeadCodeProductions/program-markers#58]
https://github.com/DeadCodeProductions/program-markers/blob/e7bf70b5034ea92b284fb7096251b2e9eb2e501e/python_src/program_markers/instrumenter.py#L304

I think the issue is all the buffering etc that is going on. I think a simple solution would be to use POpen and read stdout on the fly

**Comment by thetheodor** [comment]
```
import subprocess
import time
import threading

class SubprocessTimeout(Exception):
    pass

def target(process):
    while True:
        line = process.stdout.readline()
        if not line:
            break
        print("Read line:", line.strip())

cmd = ['ls', '-l']  # Substitute with your command

process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)

try:
    thread = threading.Thread(target=target, args=(process,))
    thread.start()

    thread.join(timeout=10)  # Timeout of 10 seconds
    if thread.is_alive():
        print('Command timeout, terminating')
        process.terminate()
        thread.join()
        raise SubprocessTimeout

except SubprocessTimeout:
    print('The subprocess did not finish in the expected time.')
finally:
    process.stdout.close()  # Ensure the subprocess gets closed properly

```

**Comment by thetheodor** [comment]
Fixed with #56 #57 #59
